From 7fb826be6e78ca2ae755063fe9953f58ad6ed5a5 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 15 Apr 2020 08:11:26 +0300
Subject: [PATCH 01/12] Revert "Enable navbar by default"

This reverts commit 09a49059a98a8302a794351c659f51d3e42863db.
---
 system.prop | 1 -
 1 file changed, 1 deletion(-)

diff --git a/system.prop b/system.prop
index 7094634..606b4f6 100644
--- a/system.prop
+++ b/system.prop
@@ -6,7 +6,6 @@ persist.sys.overlay.nightmode=true
 
 #Disable debugging strict mode toasts
 persist.sys.strictmode.disable=true
-persist.sys.phh.mainkeys=0
 
 fw.max_users=10
 persist.sys.max_profiles=10
-- 
2.17.1

