From 813461e6834c6eeea11bf10bb0c7979355052399 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 8 Feb 2020 21:58:07 +0200
Subject: [PATCH 5/7] base: add 7z, ntfs-3g & vim

	modified:   base.mk
---
 base.mk | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/base.mk b/base.mk
index 8854ea0..5a3f1ef 100644
--- a/base.mk
+++ b/base.mk
@@ -125,11 +125,11 @@ SELINUX_IGNORE_NEVERALLOWS := true
 
 # Universal NoCutoutOverlay
 PRODUCT_PACKAGES += \
-    NoCutoutOverlay
+	NoCutoutOverlay
 
 PRODUCT_PACKAGES += \
-    lightsctl \
-    uevent
+	lightsctl \
+	uevent
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/files/adbd.rc:system/etc/init/adbd.rc
@@ -158,10 +158,19 @@ PRODUCT_COPY_FILES += \
 	frameworks/native/data/etc/android.software.secure_lock_screen.xml:system/etc/permissions/android.software.secure_lock_screen.xml \
 
 PRODUCT_COPY_FILES += \
-        device/phh/treble/ld.config.26.txt:system/etc/ld.config.26.txt \
+	device/phh/treble/ld.config.26.txt:system/etc/ld.config.26.txt \
 
 PRODUCT_PACKAGES += \
-    asus-motor
+	asus-motor
+
+# additional utils
+PRODUCT_PACKAGES += \
+	7z \
+	lib7z \
+	fsck.ntfs \
+	mkfs.ntfs \
+	mount.ntfs \
+	vim
 
 # Privapp-permissions whitelist for PhhTrebleApp
 PRODUCT_COPY_FILES += \
-- 
2.25.1

