From 8d0a0f813a23cd2b0b889ca6dd27648412057257 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 25 Dec 2019 12:34:05 +0200
Subject: [PATCH 08/11] base: add "secure" file to the system image

	modified:   base.mk
---
 base.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base.mk b/base.mk
index ef30901..ec4090d 100644
--- a/base.mk
+++ b/base.mk
@@ -63,6 +63,7 @@ PRODUCT_COPY_FILES += \
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/empty:system/phh/empty \
+	device/phh/treble/empty:system/phh/secure \
 	device/phh/treble/phh-on-boot.sh:system/bin/phh-on-boot.sh
 
 PRODUCT_PACKAGES += \
-- 
2.17.1

