From a84a5e5cc5510bbf8f8a67340e1f89b78fedf601 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 25 Dec 2019 12:34:05 +0200
Subject: [PATCH 09/12] base: add "secure" file to the system image

	modified:   base.mk
---
 base.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base.mk b/base.mk
index ce318bb..02cb65c 100644
--- a/base.mk
+++ b/base.mk
@@ -64,6 +64,7 @@ PRODUCT_COPY_FILES += \
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/empty:system/phh/empty \
+	device/phh/treble/empty:system/phh/secure \
 	device/phh/treble/phh-on-boot.sh:system/bin/phh-on-boot.sh
 
 PRODUCT_PACKAGES += \
-- 
2.17.1

