From 8aae49f582bbcee3519cb586bef0953a827bbe00 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Fri, 27 Mar 2020 19:27:04 +0200
Subject: [PATCH 096/100] disable googlesans font overlay

---
 packages/overlays/Android.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index b65b8229dd8..c8721595053 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -30,7 +30,6 @@ LOCAL_REQUIRED_MODULES := \
 	FontUbuntuSourceOverlay \
 	FontComicSansSourceOverlay \
 	FontCoolstorySourceOverlay \
-	FontGoogleSansSourceOverlay \
 	FontLGSmartGothicSourceOverlay \
 	FontNotoSerifSourceOverlay \
 	FontOneplusSlateSource \
-- 
2.17.1

