From 9de03faffc5858f6f948f8cf0b0fbda91ccb3c2c Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 8 Jul 2020 15:40:08 +0300
Subject: [PATCH 3/6] use generic LatinIME in Go builds

---
 products/gms_go.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/products/gms_go.mk b/products/gms_go.mk
index 7198693..3e5f0ef 100644
--- a/products/gms_go.mk
+++ b/products/gms_go.mk
@@ -67,7 +67,6 @@ GMS_PRODUCT_PACKAGES += \
     GalleryGo \
     GMailGo \
     GoogleSearchGo \
-    LatinImeGoogleGo \
     MapsGo \
     NavGo \
     YouTube
-- 
2.17.1

