From 16317da52fa9f7da62c40ef75ea2151a56ae6fac Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Tue, 14 Aug 2018 20:54:08 +0200
Subject: [PATCH 3/9] Create vendor_de. This is done by /init.rc on
 system-as-root device

---
 FsCrypt.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/FsCrypt.cpp b/FsCrypt.cpp
index 7f2b5e4..924af98 100644
--- a/FsCrypt.cpp
+++ b/FsCrypt.cpp
@@ -676,6 +676,8 @@ bool fscrypt_prepare_user_storage(const std::string& volume_uuid, userid_t user_
         auto vendor_de_path = android::vold::BuildDataVendorDePath(user_id);
         auto user_de_path = android::vold::BuildDataUserDePath(volume_uuid, user_id);
 
+	prepare_dir(android::vold::BuildDataPath("") + "/vendor_de", 0771, 0, 0);
+
         if (volume_uuid.empty()) {
             if (!prepare_dir(system_legacy_path, 0700, AID_SYSTEM, AID_SYSTEM)) return false;
 #if MANAGE_MISC_DIRS
-- 
2.17.1

