From e0d0c7ae32fa11c5a095a6d38e853a7090012cee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Danijel=20Jovanovi=C4=87?= <danijel.xda@gmail.com>
Date: Thu, 20 Feb 2020 21:57:00 +0100
Subject: [PATCH 09/20] Fix distorted audio bug on GSI images for ginkgo and
 willow

---
 rw-system.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/rw-system.sh b/rw-system.sh
index 0a3a28c..417019e 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -282,6 +282,10 @@ if getprop ro.vendor.build.fingerprint | grep -iq -e iaomi/cactus -e iaomi/cereu
     setprop debug.stagefright.omx_default_rank 0
 fi
 
+if getprop ro.vendor.build.fingerprint | grep -iq -e xiaomi/ginkgo -e  xiaomi/willow; then
+    mount -o bind /system/phh/empty /vendor/lib/soundfx/libvolumelistener.so
+fi
+
 mount -o bind /system/phh/empty /vendor/lib/libpdx_default_transport.so
 mount -o bind /system/phh/empty /vendor/lib64/libpdx_default_transport.so
 
-- 
2.17.1

