From 7d2249f9726dd39781fa81d0b592072e6da930af Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 25 Dec 2019 12:34:05 +0200
Subject: [PATCH 07/11] base: add "secure" file to the system image

	modified:   base.mk
---
 base.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/base.mk b/base.mk
index b11fde5..e7ead35 100644
--- a/base.mk
+++ b/base.mk
@@ -65,6 +65,7 @@ PRODUCT_COPY_FILES += \
 
 PRODUCT_COPY_FILES += \
 	device/phh/treble/empty:system/phh/empty \
+	device/phh/treble/empty:system/phh/secure \
 	device/phh/treble/phh-on-boot.sh:system/bin/phh-on-boot.sh
 
 PRODUCT_PACKAGES += \
-- 
2.17.1

