From 5ea1cc31dc3a0c5fae7bcab98b085136d2722c0e Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Tue, 6 Oct 2020 14:26:50 -0400
Subject: [PATCH] add persist.sys.phh.qin.dt2w for QIN dt2w

---
 phh-prop-handler.sh | 9 +++++++++
 vndk.rc             | 3 +++
 2 files changed, 12 insertions(+)

diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 9dc9a2a..1270404 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -142,3 +142,12 @@ if [ "$1" == "persist.sys.phh.vsmart.dt2w" ];then
     fi
     exit
 fi
+
+if [ "$1" == "persist.sys.phh.qin.dt2w" ];then
+    if [[ "$prop_value" != "0" && "$prop_value" != "1" ]]; then
+        exit 1
+    fi
+
+    echo "$prop_value" > /sys/devices/platform/soc/soc:ap-apb/70800000.i2c/i2c-3/3-0038/fts_gesture_mode
+    exit
+fi
diff --git a/vndk.rc b/vndk.rc
index 9f1003a..632ef0f 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -44,6 +44,9 @@ on property:persist.sys.phh.caf.audio_policy=*
 on property:persist.sys.phh.vsmart.dt2w=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.vsmart.dt2w"
 
+on property:persist.sys.phh.qin.dt2w=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.qin.dt2w"
+
 on property:persist.sys.phh.disable_a2dp_offload=0
     setprop persist.sys.phh.disable_a2dp_offload false
 
-- 
2.17.1

