From 4f6d20fca1f7a195543d717ecee3bec2bc9a2920 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 10 Oct 2020 00:52:18 -0400
Subject: [PATCH] rework securize prop targets

---
 rw-system.sh | 103 ++++++++++++++++++++++++++-------------------------
 1 file changed, 52 insertions(+), 51 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index 336dc1b..a31b151 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -560,58 +560,59 @@ if getprop ro.vendor.build.fingerprint | grep -qiE '^samsung/' && [ "$vndk" -ge
     fi
 fi
 
-if [ -f /system/phh/secure ];then
-    copyprop() {
-        p="$(getprop "$2")"
-        if [ "$p" ]; then
-            resetprop "$1" "$(getprop "$2")"
-        fi
-    }
-
-    copyprop ro.build.device ro.vendor.build.device
-    copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.build.fingerprint ro.vendor.build.fingerprint
-    copyprop ro.build.device ro.vendor.product.device
-    copyprop ro.product.system.device ro.vendor.product.device
-    copyprop ro.product.device ro.vendor.product.device
-    copyprop ro.product.system.device ro.product.vendor.device
-    copyprop ro.product.device ro.product.vendor.device
-    copyprop ro.product.system.name ro.vendor.product.name
-    copyprop ro.product.name ro.vendor.product.name
-    copyprop ro.product.system.name ro.product.vendor.device
-    copyprop ro.product.name ro.product.vendor.device
-    copyprop ro.system.product.brand ro.vendor.product.brand
-    copyprop ro.product.brand ro.vendor.product.brand
-    copyprop ro.product.system.model ro.vendor.product.model
-    copyprop ro.product.model ro.vendor.product.model
-    copyprop ro.product.system.model ro.product.vendor.model
-    copyprop ro.product.model ro.product.vendor.model
-    copyprop ro.build.product ro.vendor.product.model
-    copyprop ro.build.product ro.product.vendor.model
-    copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
-    copyprop ro.product.manufacturer ro.vendor.product.manufacturer
-    copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
-    copyprop ro.product.manufacturer ro.product.vendor.manufacturer
-    copyprop build.version.security_patch ro.build.version.security_patch
-    (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
-        [ -n "$v" ] && resetprop ro.build.version.security_patch "$v"
-    done
+copyprop() {
+  p="$(getprop "$2")"
+  if [ "$p" ]; then
+    resetprop "$1" "$(getprop "$2")"
+  fi
+}
+
+copyprop ro.build.device ro.vendor.build.device
+copyprop ro.system.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.bootimage.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.build.fingerprint ro.vendor.build.fingerprint
+copyprop ro.build.device ro.vendor.product.device
+copyprop ro.product.system.device ro.vendor.product.device
+copyprop ro.product.device ro.vendor.product.device
+copyprop ro.product.system.device ro.product.vendor.device
+copyprop ro.product.device ro.product.vendor.device
+copyprop ro.product.system.name ro.vendor.product.name
+copyprop ro.product.name ro.vendor.product.name
+copyprop ro.product.system.name ro.product.vendor.device
+copyprop ro.product.name ro.product.vendor.device
+copyprop ro.system.product.brand ro.vendor.product.brand
+copyprop ro.product.brand ro.vendor.product.brand
+copyprop ro.product.system.model ro.vendor.product.model
+copyprop ro.product.model ro.vendor.product.model
+copyprop ro.product.system.model ro.product.vendor.model
+copyprop ro.product.model ro.product.vendor.model
+copyprop ro.build.product ro.vendor.product.model
+copyprop ro.build.product ro.product.vendor.model
+copyprop ro.system.product.manufacturer ro.vendor.product.manufacturer
+copyprop ro.product.manufacturer ro.vendor.product.manufacturer
+copyprop ro.system.product.manufacturer ro.product.vendor.manufacturer
+copyprop ro.product.manufacturer ro.product.vendor.manufacturer
 
-    resetprop ro.build.tags release-keys
-    resetprop ro.boot.vbmeta.device_state locked
-    resetprop ro.boot.verifiedbootstate green
-    resetprop ro.boot.flash.locked 1
-    resetprop ro.boot.veritymode enforcing
-    resetprop ro.boot.warranty_bit 0
-    resetprop ro.warranty_bit 0
-    resetprop ro.debuggable 0
-    resetprop ro.secure 1
-    resetprop ro.build.type user
-    resetprop ro.build.selinux 0
-
-    resetprop ro.adb.secure 1
-    setprop ctl.restart adbd
+if [ -f /system/phh/secure ];then
+  copyprop build.version.security_patch ro.build.version.security_patch
+  (getprop ro.vendor.build.security_patch; getprop ro.keymaster.xxx.security_patch) |sort |tail -n 1 |while read v;do
+    [ -n "$v" ] && resetprop ro.build.version.security_patch "$v"
+  done
+
+  resetprop ro.build.tags release-keys
+  resetprop ro.boot.vbmeta.device_state locked
+  resetprop ro.boot.verifiedbootstate green
+  resetprop ro.boot.flash.locked 1
+  resetprop ro.boot.veritymode enforcing
+  resetprop ro.boot.warranty_bit 0
+  resetprop ro.warranty_bit 0
+  resetprop ro.debuggable 0
+  resetprop ro.secure 1
+  resetprop ro.build.type user
+  resetprop ro.build.selinux 0
+
+  resetprop ro.adb.secure 1
+  setprop ctl.restart adbd
 fi
 
 for abi in "" 64;do
-- 
2.17.1

