From 65a459c896cf5b66371ac5fefde8d1611b376021 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 9 Apr 2020 05:32:23 +0300
Subject: [PATCH 08/10] add a lot of fonts v217

---
 target/product/handheld_system.mk | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index 5ebceeafe..a0ed1ae78 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -32,6 +32,24 @@ $(call inherit-product-if-exists, external/google-fonts/rubik/fonts.mk)
 $(call inherit-product-if-exists, external/google-fonts/zilla-slab/fonts.mk)
 $(call inherit-product-if-exists, external/noto-fonts/fonts.mk)
 $(call inherit-product-if-exists, external/roboto-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/ubuntu-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/alegreyasans-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/arsenal-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/caveat-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/comfortaa-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/cormorantinfant-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/cuprum-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/exo2-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/gabriela-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/inter-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/jura-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/lobster-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/montserrat-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/oswald-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/pacifico-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/philosopher-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/viaodalibre-fonts/fonts.mk)
+$(call inherit-product-if-exists, external/vollkorn-fonts/fonts.mk)
 $(call inherit-product-if-exists, external/hyphenation-patterns/patterns.mk)
 $(call inherit-product-if-exists, frameworks/base/data/keyboards/keyboards.mk)
 $(call inherit-product-if-exists, frameworks/webview/chromium/chromium.mk)
-- 
2.17.1

