From 66b81c09de96bfd75b5a4209c8e4b38a0e1448d0 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Wed, 8 Jul 2020 15:40:08 +0300
Subject: [PATCH 3/3] use generic LatinIME in Go builds

---
 products/gms_go.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/products/gms_go.mk b/products/gms_go.mk
index 55f445a..8d33151 100644
--- a/products/gms_go.mk
+++ b/products/gms_go.mk
@@ -67,7 +67,6 @@ GMS_PRODUCT_PACKAGES += \
     GalleryGo \
     GMailGo \
     GoogleSearchGo \
-    LatinImeGoogleGo \
     MapsGo \
     NavGo \
     YouTubeGo
-- 
2.25.1

