From 27b57beac2d9ad2eae0200724ad86331b2f92d18 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 15 Aug 2020 21:42:26 -0400
Subject: [PATCH 7/7] add face id

---
 config/common.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/config/common.mk b/config/common.mk
index 1bde29df..d867e1ea 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -158,6 +158,9 @@ PRODUCT_PACKAGES += \
 PRODUCT_PACKAGES += \
     rsync
 
+PRODUCT_PACKAGES += \
+    FaceUnlockService
+
 # Storage manager
 PRODUCT_SYSTEM_DEFAULT_PROPERTIES += \
     ro.storage_manager.enabled=true
-- 
2.17.1

